CMakeLists.txt

```diff
@@ -441,14 +441,26 @@ if (WIN32)
 endif (WIN32)
 
 if (UNIX AND NOT PIONEER_INSTALL_INPLACE)
-	set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/metadata/net.pioneerspacesim.Pioneer.desktop)
-	configure_file(metadata/net.pioneerspacesim.Pioneer.desktop.cmakein ${PIONEER_DESKTOP_FILE} @ONLY)
-	install(FILES ${PIONEER_DESKTOP_FILE}
-		DESTINATION ${CMAKE_INSTALL_DATADIR}/applications
-	)
-	install(FILES metadata/net.pioneerspacesim.Pioneer.appdata.xml
-		DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata
-	)
+    if(APPIMAGE_BUILD)
+        set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/appimage_metadata/net.pioneerspacesim.Pioneer.desktop)
+        set(PIONEER_SH_FILE ${CMAKE_BINARY_DIR}/init_pioneer.sh)
+        configure_file(appimage_metadata/net.pioneerspacesim.Pioneer.desktop.cmakein ${PIONEER_DESKTOP_FILE} @ONLY)
```

Comment:
The `PIONEER_DESKTOP_FILE` variable and the configure_file() can use the files in `metadata/` like I said in my other comment (with a small update to the .desktop.cmakein file), so this part of the CMake script wouldn't need to change and the diff can be made much smaller.